pykafka/protocol.py

```diff
@@ -172,10 +172,13 @@ def __init__(self,
         self.produce_attempt = produce_attempt
 
     def __len__(self):
-        size = 4 + 1 + 1 + 4 + 4 + len(self.value)
-        if self.partition_key is not None:
-            size += len(self.partition_key)
-        return size
+        if not self.value==None:
+            size = 4 + 1 + 1 + 4 + 4 + len(self.value)
+            if self.partition_key is not None:
+                size += len(self.partition_key)
+            return size
+        else:
+            return 0
```

Comment:
Below code will consider everything except for `None` length
`size = 4 + 1 + 1 + 4 + 4
if self.partition_key is not None:
      size += len(self.partition_key)
return size`